go/exocom/src/exocom/exocom.go

```diff
@@ -106,3 +125,71 @@ func parseRegisterMessagePayload(message structs.Message) (string, error) {
 	}
 	return "", fmt.Errorf("Invalid register message payload: %v", message.Payload)
 }
+
+func (e *ExoCom) send(message structs.Message) error {
+	publicMessageName := messageTranslator.GetPublicMessageName(&messageTranslator.GetPublicMessageNameOptions{
+		Namespace:           e.clientRegistry.Clients[message.Sender].InternalNamespace,
+		ClientName:          message.Sender,
+		InternalMessageName: message.Name,
+	})
+	originalName := message.Name
+	message.Name = publicMessageName
+	message.Timestamp = time.Now()
+	subscribers := e.clientRegistry.GetSubscribersFor(publicMessageName)
+	if len(subscribers) == 0 {
+		printLogError(e.logger.Warning(fmt.Sprintf("Warning: No receivers for message '%s' registered", publicMessageName)))
+		return nil
+	}
+	if message.ResponseTo != "" {
+		originalTimestamp, ok := e.messageCache.Get(message.ID)
+		if ok {
+			message.ResponseTime = message.Timestamp.Sub(originalTimestamp)
+		}
+	} else {
+		e.messageCache.Set(message.ID, message.Timestamp)
+	}
+	internalMessageNameMapping, err := e.sendToServices(message, subscribers)
+	if err != nil {
+		return err
+	}
+	printLogError(e.logger.Messages(message, originalName, internalMessageNameMapping))
+	return nil
+}
+
+func (e *ExoCom) sendToServices(message structs.Message, subscribers []clientRegistry.Subscriber) (map[string]string, error) {
```

Comment:
What sort of map does this method return? Ideally this is documented via a variable name for the result map.